### PR TITLE
Exclude slow tests from results completely

### PR DESCRIPTION
**Pull request by angelikatyborska**
https://github.com/exercism/elixir-test-runner/pull/47 was a really good PR, but we didn't realize that it will lead to slow tests being marked as failing on the platform. We need to exclude the slow tests from results completely.

Screen Shot 2021-07-30 at 19 01 29

### CODE REVIEW

**Comment by angelikatyborska** [comment]
🤔 wrong approach

**Comment by angelikatyborska** [comment]
I'm not sure if it's better to explicitly exclude all tests tagged `@slow` like now, or to hijack the test formatter (https://github.com/exercism/elixir-test-runner/blob/bef1aa0bb1520732809353479e89b61e3fd09af8/exercism_test_helper/lib/json_formatter.ex#L86-L101) and treat all exunit-level-excluded tests as excluded.

**Comment by jiegillet** [comment]
Would it be as simple as this?
```diff
- test = %{name: test.name, status: :fail, message: "test excluded"} 
+ test = %{name: test.name, status: :pass, message: "test excluded"} 
```
If so, I'm partial to it :)
(disclaimer: I have not looked very deep into this at all!)

**Comment by angelikatyborska** [comment]
No, not really _that_ simple. If we decided to change how excluded tests are treated, I would say they should not be reported as passing, but rather completely excluded from the results.

But I am not sure if that's a good decision. If the test is excluded for unexpected reasons, maybe the test suite should fail?

**Review by neenjaw — APPROVED** [review]
👍

File: test/expected_results.json [hunk 1]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+{"message":null,"status":"fail","tests":[{"message":null,"name":"test says 'Hello, World!'","output":"","status":"pass","task_id":1,"test_code":"assert HelloWorld.hello == \"Hello, World!\""},{"message":"  1) test failing test (HelloWorldTest)\n     \u001B[1m\u001B[30mtest/hello_world_test.exs:9\u001B[0m\n     \u001B[31mAssertion with == failed\u001B[0m\n     \u001B[36mcode:  \u001B[0massert [1, 2, 3] == [2, 3]\n     \u001B[36mleft:  \u001B[0m[\u001B[31m1\u001B[0m, 2, 3]\n     \u001B[36mright: \u001B[0m[2, 3]\n     \u001B[36mstacktrace:\u001B[0m\n       test/hello_world_test.exs:11: (test)\n","name":"test failing test","output":"","status":"fail","task_id":2,"test_code":"assert [1, 2, 3] == [2, 3]"}],"version":3}
```
Comment (jiegillet) [inline test/expected_results.json:1]: We should have caught it here
```
{"message":null,"status":"fail","tests":[...]}
```

**Pull request merged**